tests/backend/test_validators.py

```diff
@@ -0,0 +1,296 @@
+"""Tests for validators for DatadocMetadata class."""
+
+from __future__ import annotations
+
+import datetime
+import re
+import warnings
+from typing import TYPE_CHECKING
+
+import datadoc_model
+import pytest
+from datadoc_model import model
+from pydantic import ValidationError
+
+from datadoc import state
+from datadoc.backend.model_validation import ObligatoryDatasetWarning
+from datadoc.backend.model_validation import ObligatoryVariableWarning
+from datadoc.backend.utils import incorrect_date_order
+from datadoc.enums import TemporalityTypeType
+
+if TYPE_CHECKING:
+    from datadoc.backend.core import Datadoc
+
+
+def test_incorrect_date_order(metadata: Datadoc):
+    variables = metadata.variables
+    dataset = metadata.dataset
+    assert variables is not None
+    variables[0].contains_data_from = datetime.date(2024, 1, 1)
+    variables[0].contains_data_until = datetime.date(1960, 1, 1)
+    result_incorrect_dates = incorrect_date_order(
+        variables[0].contains_data_from,
+        variables[0].contains_data_until,
+    )
+    assert result_incorrect_dates is True
+    dataset.contains_data_from = datetime.date(1980, 1, 1)
+    dataset.contains_data_until = datetime.date(2000, 6, 5)
+    result_correct_dates = incorrect_date_order(
+        dataset.contains_data_from,
+        dataset.contains_data_until,
+    )
+    assert result_correct_dates is False
```

Comment:
This test case should be extracted with parameterize. Would be good to have a test case for when the date is identical too.